docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+        from("$reactAndroidProjectDir/src/main/jni/prebuilt/lib")
+        into("$buildDir/react-ndk/exported")
+    }
+
+    afterEvaluate {
+        preBuild.dependsOn(packageReactNdkLibs)
+    }
+
+    packagingOptions {
+        pickFirst '**/libhermes.so'
+        pickFirst '**/libjsc.so'
+    }
+}
+```
+
+Finally, we need to create a Makefile inside the `src/main/jni` folder called `Android.mk` with the following content:
+
+```makefile
+THIS_DIR := $(call my-dir)
+
+include $(REACT_ANDROID_DIR)/Android-prebuilt.mk
+include $(GENERATED_SRC_DIR)/codegen/jni/Android.mk
+
+include $(CLEAR_VARS)
+
+LOCAL_PATH := $(THIS_DIR)
+LOCAL_MODULE := myapplication_appmodules
+
+LOCAL_C_INCLUDES := $(LOCAL_PATH) $(GENERATED_SRC_DIR)/codegen/jni
+LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp) $(wildcard $(GENERATED_SRC_DIR)/codegen/jni/*.cpp)
+LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH) $(GENERATED_SRC_DIR)/codegen/jni
+
+# Please note as one of the library listed is libreact_codegen_samplelibrary
+# This name will be generated as libreact_codegen_<library-name>
+# where <library-name> is the one you specified in the Gradle configuration
+LOCAL_SHARED_LIBRARIES := libjsi libfbjni libglog libfolly_json libyoga libreact_nativemodule_core libturbomodulejsijni librrc_view libreact_render_core libreact_render_graphics libfabricjni libfolly_futures libreact_debug libreact_render_componentregistry libreact_render_debug libruntimeexecutor libreact_codegen_rncore libreact_codegen_samplelibrary
+
+LOCAL_CFLAGS := \
+    -DLOG_TAG=\"ReactNative\"
+LOCAL_CFLAGS += -fexceptions -frtti -std=c++17 -Wall
+
+include $(BUILD_SHARED_LIBRARY)
+```
+
+This setup will run a native build on your project and will compile the C++ files that have been generated by the codegen. You will see the native build running with the Gradle task `:app:externalNativeBuildRelease`
+
+You can now verify that everything works correctly by running your android app:
+
+```bash
+yarn react-native run-android
+```
+
+### 2. Java - Provide a `ReactPackageTurboModuleManagerDelegate`
+
+Now is time to actually use the TurboModule.
+First, we will need to create a `ReactPackageTurboModuleManagerDelegate` subclass, like the following:
+
+```java
+package com.awesomeproject;
+
+import com.facebook.jni.HybridData;
+import com.facebook.react.ReactPackage;
+import com.facebook.react.ReactPackageTurboModuleManagerDelegate;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.soloader.SoLoader;
+
+import java.util.List;
+
+public class MyApplicationTurboModuleManagerDelegate extends ReactPackageTurboModuleManagerDelegate {
+
+    private static volatile boolean sIsSoLibraryLoaded;
+
+    protected MyApplicationTurboModuleManagerDelegate(ReactApplicationContext reactApplicationContext, List<ReactPackage> packages) {
+        super(reactApplicationContext, packages);
+    }
+
+    protected native HybridData initHybrid();
+
+    native boolean canCreateTurboModule(String moduleName);
```

Comment:
I don't think this method is actually needed. Maybe we can remove it for simplicity.

Reply:
Yup you're right, removed.